htlcswitch/quiescer.go

```diff
@@ -292,6 +337,24 @@ func (q *quiescer) resume() {
 	q.resumeQueue = nil
 }
 
+type quiescerNoop struct{}
+
+var _ quiescer = (*quiescerNoop)(nil)
```

Comment:
Regarding "test the noop quiescer" and "unification for the main quiescer vs the noop quiescer", I'm assuming you mean testing the behavior of the `quiescer` in the link as in how it's used? I think the easiest way is to use `mock.Mock` and check the methods are called as expected. With mock we can easily imitate the behavior of either `quiescer` implementation.

If you mean itest I think the `quiescerNoop` is already been tested since it's the default setting.